pkg/reconciler/v1alpha1/pipelinerun/pipelinerun_test.go

```diff
@@ -419,13 +435,296 @@ func TestUpdateTaskRunsState(t *testing.T) {
 
 }
 
+// TODO: Switch to single test func with diff inputs
```

Comment:
tbh it feels a bit easier to read the way you have it! but i could be convinced otherwise :)

reconciler level tests are a bit hard to follow anyway :S the more we can avoid testing at this level the better!

Reply:
yeah so this comment is from a previous review; I did attempt to make this into a single function but I don't think it was particularly readable - especially the part where we need to add nested maps. I think I'll just remove the TODO